### PR TITLE
SLE-18 Add NOSONAR quick fix

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/NoSonarResolution.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  @Override
+  public Image getImage() {
+    return SonarLintImages.SONAR16_IMG;
+  }
+
+  @Override
+  public IMarker[] findOtherMarkers(IMarker[] markers) {
+    List<IMarker> result = new ArrayList<>();
+    for (IMarker iMarker : markers) {
+      if (iMarker != this.marker) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/NoSonarResolution.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/NoSonarResolution.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      final MultiTextEdit edit = new MultiTextEdit();
+      edit.addChild(new InsertEdit(endOfLine, " " + NOSONAR_COMMENT));
+      edit.apply(iDoc);
+      marker.delete();
+    } catch (Exception e) {
+      SonarLintCorePlugin.getDefault().error("Unable to apply NOSONAR resolution: " + e.getMessage(), e);
+    } finally {
+      try {
+        iTextFileBufferManager.disconnect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+      } catch (CoreException e) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/NoSonarResolution.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/MarkerResolutionGenerator.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      || marker.getAttribute(IMarker.CHAR_END, 0) > 0;
+  }
+
+  private boolean isOnFile(final IMarker marker) {
+    return marker.getResource().getAdapter(IFile.class) != null;
+  }
+
+  @Override
+  public IMarkerResolution[] getResolutions(final IMarker marker) {
+    return hasResolutions(marker) ? new IMarkerResolution[] {new NoSonarResolution(marker)} : new IMarkerResolution[0];
```
Comment (SonarTech) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/resolver/MarkerResolutionGenerator.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 1 critical
* MAJOR 4 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonar-eclipse/blob/66803c215f3c749343f847443325c6119da090b6/org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java#L64) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)
* MAJOR Add a 'package-info.java' file to document the 'resolver' package (org.sonarsource.sonarlint-eclipse:org.sonarlint.eclipse.ui:src/org/sonarlint/eclipse/ui/internal/resolver) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 1 critical
* MAJOR 4 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonar-eclipse/blob/66803c215f3c749343f847443325c6119da090b6/org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java#L64) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)
* MAJOR Add a 'package-info.java' file to document the 'resolver' package (org.sonarsource.sonarlint-eclipse:org.sonarlint.eclipse.ui:src/org/sonarlint/eclipse/ui/internal/resolver) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 1 critical
* MAJOR 4 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonar-eclipse/blob/66803c215f3c749343f847443325c6119da090b6/org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java#L64) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)
* MAJOR Add a 'package-info.java' file to document the 'resolver' package (org.sonarsource.sonarlint-eclipse:org.sonarlint.eclipse.ui:src/org/sonarlint/eclipse/ui/internal/resolver) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**